File: app/src/main/java/com/datarockets/mnchkn/fragments/charts/ChartsInteractorImpl.java
package com.datarockets.mnchkn.fragments.charts;

import android.content.Context;

import com.datarockets.mnchkn.MunchkinApplication;
import com.datarockets.mnchkn.activities.BaseInteractor;
import com.datarockets.mnchkn.store.GameService;
import com.datarockets.mnchkn.store.PlayerService;

import javax.inject.Inject;

import lecho.lib.hellocharts.model.LineChartData;

public class ChartsInteractorImpl extends BaseInteractor implements ChartsInteractor {

    @Inject
    GameService mGameService;
    @Inject
    PlayerService mPlayerService;

    public ChartsInteractorImpl(Context context) {
        super(context);
    }

    @Override
    protected void setUpComponent(Context context) {
        MunchkinApplication.get(context).getAppComponent().inject(this);
    }

    @Override
    public LineChartData loadLineChartData(int type) {
        return mGameService.createScoresChartData(type, mGameService.getScoresChartData());
    }

    @Override
    public void loadPlayers(int type, OnChartLoadedListener listener) {
        listener.showPlayers(mPlayerService.getPlayersList(type));
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/charts/ChartsInteractorImpl.kt
package com.datarockets.mnchkn.fragments.charts

import android.content.Context
import com.datarockets.mnchkn.MunchkinApplication
import com.datarockets.mnchkn.activities.BaseInteractor
import com.datarockets.mnchkn.store.GameService
import com.datarockets.mnchkn.store.PlayerService
import lecho.lib.hellocharts.model.LineChartData
import javax.inject.Inject

class ChartsInteractorImpl(context: Context) : BaseInteractor(context), ChartsInteractor {

    @Inject
    lateinit var mGameService: GameService
    @Inject
    lateinit var mPlayerService: PlayerService

    override fun setUpComponent(context: Context) {
        MunchkinApplication.get(context).appComponent.inject(this)
    }

    override fun loadLineChartData(type: Int): LineChartData {
        return mGameService.createScoresChartData(type, mGameService.scoresChartData)
    }

    override fun loadPlayers(type: Int, listener: ChartsInteractor.OnChartLoadedListener) {
        listener.showPlayers(mPlayerService.getPlayersList(type))
    }

}
